packages/ckeditor5-coremedia-blocklist/src/blocklistui.ts

```diff
@@ -0,0 +1,443 @@
+    // For clicks on a blocklisted word, no value is set since the selection is collapsed
+    this.blocklistActionsView.blocklistInputView.setInputText(this.#getSelectedText());
+
+    this.#addBalloonView();
+
+    if (forceFocus) {
+      this.blocklistActionsView.focus();
+    }
+  }
+
+  /**
+   * Closes the blocklist actions view in the contextual balloon.
+   * If another view is present, that one will be shown instead.
+   *
+   * @private
+   */
+  #hideBlocklistBalloon(): void {
+    const editor = this.editor;
+
+    this.stopListening(editor.ui, "update");
+    this.stopListening(this.#balloon, "change:visibleView");
+
+    // Make sure the focus always gets back to the editable _before_ removing the focused form view.
+    // Doing otherwise causes issues in some browsers. See https://github.com/ckeditor/ckeditor5-link/issues/193.
+    editor.editing.view.focus();
+
+    // Blur the input element before removing it from DOM to prevent issues in some browsers.
+    // See https://github.com/ckeditor/ckeditor5/issues/1501.
+    if (this.blocklistActionsView) {
+      this.blocklistActionsView.blocklistInputView.saveButtonView.focus();
+    }
+    // Then remove the blocklist.
+    if (this.#balloon && this.blocklistActionsView) {
+      this.#balloon.remove(this.blocklistActionsView);
+    }
+
+    // reset the command value
+    this.blocklistCommand?.set("value", []);
+  }
+
+  /**
+   * Adds the blocklist view to the balloon views.
+   * This will result in opening the balloon, if it was not visible until now.
+   *
+   * @private
+   */
+  #addBalloonView(): void {
+    if (!this.blocklistActionsView) {
+      this.#createBlocklistActionsView();
+    }
+
+    if (!this.#balloon) {
+      return;
+    }
+
+    this.#balloon.add({
+      view: this.blocklistActionsView,
+      position: this.#getBalloonPositionData(),
+    });
+  }
+
+  #getBalloonPositionData(): Partial<PositionOptions> {
+    const view = this.editor.editing.view;
+    const viewDocument = view.document;
+
+    const selection = view.document.selection;
+    const selectedElement = selection.getSelectedElement();
+    const target: PositionOptions["target"] = () => {
+      const targetWord = selectedElement;
+
+      if (targetWord) {
+        const targetWordViewElement = view.domConverter.mapViewToDom(targetWord);
+        if (targetWordViewElement) {
+          return targetWordViewElement;
+        }
+      }
+
+      // Otherwise attach panel to the selection.
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      return view.domConverter.viewRangeToDom(viewDocument.selection.getFirstRange()!);
+    };
+    return { target };
+  }
+
+  /**
+   * Initializes listeners to open the balloon, or switch to the blocklist balloon view.
+   * The balloon opens, if the command is enabled and the shortcut is pressed,
+   * or if a block-listed word is clicked on.
```

Comment:
```suggestion
   * or if a blocklisted word is clicked on.
```

In other comments you use `blocklisted word` instead of `block-listed word`. 
I would be fine with `blocked word` too